server/request_sendrawtx.go

```diff
@@ -120,7 +120,7 @@ func (r *RpcRequest) handle_sendRawTransaction() {
 	if r.tx.To() != nil && len(r.tx.Data()) <= 2 && txFromLower == strings.ToLower(r.tx.To().Hex()) {
 		requestDone := r.handleCancelTx() // returns true if tx was cancelled at the relay and response has been sent to the user
 		r.ethSendRawTxEntry.IsCancelTx = true
-		if requestDone {
+		if requestDone && r.preferences == nil { // if tx was cancelled and fast endpoint was not called
```

Comment:
- No way it would be false if it contains preference object. 

- Do you want to change it like this ? 

`if requestDone {
			if r.preferences != nil && !r.preferences.Fast { // if tx was cancelled and fast endpoint was not called
				return
			}
		}`

or  

- you want the preference object not to be a pointer ? Because if we chane the preference as not be a pointer then what you mentioned can be done  as below.

`if requestDone  && !r.preferences.Fast { // if tx was cancelled and fast endpoint was not called
				return
			}
		}`
But I don't see any valid reason to do like this

Reply:
Imagine we add more preferences... then it's easy to forget about this line! And it will break.

I think every request should have the preferences object, rather than only create it when `fast` is used. The no-pointer option is better i think (just because there's less possibility of nil), but the other one also works and we can use that if you prefer (note that in your first example it should be `r.preferences == nil || !r.preferences.Fast`)

Reply:
Ok I will proceed with no pointer preferences option.

Reply:
we could do like this also.
```go
preferences := new(types.PrivateTxPreferences)
````